Enable contributors developing on Apple Silicon (M1 - Arm64)

LanceEa opened the pull request:
## Description
The Emissary-Ingress build system supports and tests 'linux/amd64' containers which presents a problem when developing locally on newer M1 Macs which have Apple Silicon (arm64). To enable developers to build locally on Apple Silicon we have introduce a new environment variable `BUILD_ARCH` which defaults to `linux/amd64` so that running `make images` will produce the correct artifacts.

**Important** - This does NOT introduce a multi-architecture image but rather to decrease the friction for contributors on Apple Silicon. The multi-architecture support is a separate stream which some of the discussion can be found in #4184.


## Related Issues
N/A

## Testing

- Manually verified that `make images` is able to properly build the containers
- Manually verified that `make deploy` can push containers out to development K8s cluster. 
- Verified that Emissary-Ingress ran successfully within cluster after forcing linux/amd64 architecture
- Tested that Make variable `BUILD_ARCH` defaults `linux/amd64` but can be overridden by setting the Environment Variable with the same name

## Checklist

<!--
  Please review the requirements for each checkbox, and check them
  off (change "[ ]" to "[x]") as you verify that they are complete.
-->

 - [ ] I made sure to update `CHANGELOG.md`.
   
I haven't because this doesn't really apply to a specific release...however, it might be good to cherry pick these changes to any release branch that is still supported?
 
 - [x] This is unlikely to impact how Ambassador performs at scale.
 
 - [x] My change is adequately tested.

It might be good if someone without an M1 Mac or another developer on the M1 Mac tries it out as well but so far there doesn't appear to be any issues with using this approach in the short-term.
 
 - [] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

LanceEa commented:
@kflynn @LukeShu 

1. Are you cool with the variable name I introduced and location or is their a more preferred location?
2. Should I update the `/docs/releaseNotes.yml` since this is more of a contributor facing PR?
3. Do we cherry-pick this kind of stuff into release branches?

kflynn reviewed (changes requested):
@LanceEa Looks good from here! One request: please drop a note into `DEVELOPING.md` about this? but yeah, I'm good with the content. (I would've been OK with `BUILD_ARCH` for the name as well, but I'm good with `CONTAINER_BUILD_ARCH`.)

AliceProxy commented:
@LanceEa Normally before the next release we will create the release branch off of master so landing that change here is fine. We normally wouldn't cherry-pick it into release branches such as 2.1 that have already been released since the state of those branches should be everything that was in that release.

LanceEa commented:
> @LanceEa Looks good from here! One request: please drop a note into `DEVELOPING.md` about this? but yeah, I'm good with the content. (I would've been OK with `BUILD_ARCH` for the name as well, but I'm good with `CONTAINER_BUILD_ARCH`.)

I originally had `BUILD_ARCH` so since you will accept that I changed it to keep it smaller 😄 . I have added a few changes to `DEVELOPING.md` so let me know if that works for you.

AliceProxy reviewed (approved):
Awesome stuff :raised_hands:

kflynn reviewed (approved):
Looks good, hit it! 🙂
